### ISSUE TITLE
Rancher agent (network manager) failed to start after reboot

### ISSUE THREAD

**Issue by panho66** [rancher/rancher#8582]
**Rancher Versions:**
Server: 1.5.5
healthcheck: v0.2.3
ipsec:
network-services: v0.6.6
scheduler:
kubernetes (if applicable):

**Docker Version:**
Docker version 1.12.6, build 96d83a5/1.12.6
**OS and where are the hosts located? (cloud, bare metal, etc):**
Red Hat Enterprise Linux Server release 7.3 (Maipo)
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single Node rancher server
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
Install Rancher server
 docker run -d --restart=unless-stopped -p 8080:8080 rancher/server:v1.5.5
Install Rancher agent in other host
Copy and paste command from Add Host Rancher UI
After Rancher agent start and able to see host information in Rancher Server UI with all infrastructure stack container up and running including network-manager.
In Rancher agent host, reboot
shutdown -r now
**Results:** 
Rancher agent network-manager failed to start 
Found error message from rancher network manager

container_linux.go:247: starting container process caused "process_linux.go:359: container init caused \"rootfs_linux.go:54: mounting \\\"/var/lib/docker/containers/e1410808cb22e62f2a4fc40abe5a4c2f7a10f70e27935856c0cfd0706385a02c/secrets\\\" to rootfs \\\"/var/lib/docker/devicemapper/mnt/9a30da0c77d5893dddadecf60922b41c8b09d2ed555d253d3ae45da2bf76b55b/rootfs\\\" at \\\"/var/lib/docker/devicemapper/mnt/9a30da0c77d5893dddadecf60922b41c8b09d2ed555d253d3ae45da2bf76b55b/rootfs/run/secrets\\\" caused \\\"mkdir /var/lib/docker/devicemapper/mnt/9a30da0c77d5893dddadecf60922b41c8b09d2ed555d253d3ae45da2bf76b55b/rootfs/run/secrets: read-only file system\\\"\""

![image](https://cloud.githubusercontent.com/assets/15118220/25324334/a52947c0-2908-11e7-9257-8e33f5abc177.png)

**Comment by tfiduccia** [comment]
@panho66 - I am unable to reproduce your problem.  Using rancher v1.5.5, I created a rhel machine with docker 17.04.0-ce through AWS and after rebooting everything came back up.

**Comment by panho66** [comment]
I am using docker 1.12.6 which is only RedHat supported version so far  :( and I use devicemapper which is recommended by ReHat.

**Comment by aleksejlopasov** [comment]
@panho66 - Also using redhat/centos machines but changed docker (from redhat repo) + devicemapper to official docker + overlay (by the way v17.03 is stable version and uses overlay storage driver by default). We had to do it due to issue https://github.com/moby/moby/issues/28754. Overlay is simpler and faster than devicemapper.
Although if you want to use devicemapper only, please make sure you use direct-lvm + XFS mode instead of default loop + EXT4 mode.

**Comment by panho66** [comment]
It works after I switch over from Redhat repos to Centos Repos installing Docker-Engine 1.12.6-1 instead of Docker 1.12.6-11 from RatHat Repo.

Any idea the different ? I still using devicemapper although it is slower but got RatHat support. Overlay storage driver is support with restriction. :(

**Docker Info**

Containers: 9
 Running: 9
 Paused: 0
 Stopped: 0
Images: 13
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: datavg-docker_pool
 Pool Blocksize: 524.3 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: xfs
 Data file:
 Metadata file:
 Data Space Used: 2.074 GB
 Data Space Total: 106.3 GB
 Data Space Available: 104.2 GB
 Metadata Space Used: 1.171 MB
 Metadata Space Total: 2.143 GB
 Metadata Space Available: 2.142 GB
 Thin Pool Minimum Free Space: 10.63 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-514.6.2.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.64 GiB
Name: testing.corp
ID: XXI2:KOEN:M5AR:N7OU:NS6J:OXMV:VB5L:ET22:25YJ:U5VA:ZFHX:Z5H5
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Http Proxy: http://proxy.corp:8080
Https Proxy: http://proxy.corp:8080
No Proxy: 127.0.0.1,localhost,.corp,10.138.145.231,testing
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8

**Comment by deniseschannon** [comment]
There are some known issues with CentOS and the latest versions due to https://github.com/rancher/rancher/issues/6111

If you had no issues with RedHat and moved to CentOS, that would be the most likely root cause. 

We should have a fix released for v1.6.

**Comment by panho66** [comment]
@deniseschannon the problem is it works for Centos repos but not Redhat repos. I have raised ticket to Redhat to find out more information :)

**Comment by aemneina** [comment]
@panho66 you might also want to try to see if disabling selinux allows you to move forward with RHEL. @joshwget upstreamed a fix to selinux to grant the correct permissions w/ RHEL+selinux for docker. so this might be worth checking out.

**Comment by themoses** [comment]
Any news on this? We are currently experiencing the same issues using SLES 12 SP2 with the latest supported docker version 1.12.6-90 in the SUSE Repos in combination with Rancher 1.6.0

**Comment by panho66** [comment]
@themoses It seems there is a bug in Docker in Redhat repos's version. Docker-Engine is OK in centos repos. RedHat has  open an internal bug in bugzilla which I can't see and Redhat will keep me post every 15 days :(  and was told  working on bugzilla usually takes quite some time, So, I move on to use Docker-Engine in centos repos. (docker-engine.x86_64    1.12.6-1.el7.centos). I will post information here if I got any from Redhat.

**Comment by runcom** [comment]
@ibuildthecloud is there any reason why the network-manager container run with `/run:/run:ro` bind mounted? `/run` is supposed to be writable like `/tmp` and that is breaking network-manager on RHEL (we mount RHEL secrets in containers at `/run/secrets` and with read-only `/run` that fails. Can you guys get rid of that read-only bind?

**Comment by runcom** [comment]
Any update on this guys?

**Comment by themoses** [comment]
We managed to work around this using the docker for the enterprise repo which use a more up to date docker version. Not really an update but rather a confirmation of the bug...

**Comment by sangeethah** [comment]
Tested with rancher-server version - v1.6.3-rc5

Added custom host to the rancher-server setup with following configuration:
rhel 7.3 with  native docker 1.12.6 installed.

Docker was installed using the following steps:
```
sudo yum-config-manager --enable rhui-REGION-rhel-server-extras
sudo yum install -y docker-2:1.12.6-32.git88a4867.el7.x86_64
```
Docker uses device mapper:

```
[root@ip-172-31-8-229 ec2-user]# docker info
Containers: 12
 Running: 12
 Paused: 0
 Stopped: 0
Images: 8
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 1.411 GB
 Data Space Total: 204 GB
 Data Space Available: 202.6 GB
 Metadata Space Used: 1.04 MB
 Metadata Space Total: 2.143 GB
 Metadata Space Available: 2.142 GB
 Thin Pool Minimum Free Space: 20.4 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: overlay bridge null host
 Authorization: rhel-push-plugin
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Security Options: seccomp selinux
Kernel Version: 3.10.0-514.26.1.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 1
Total Memory: 991.2 MiB
Name: ip-172-31-8-229.us-west-2.compute.internal
ID: GMZM:7RLS:O26R:6ZP7:FQG3:OFOG:22B7:6FNR:X47Y:WITB:LZAC:IS6T
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://registry.access.redhat.com/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
Registries: registry.access.redhat.com (secure), docker.io (secure)
[root@ip-172-31-8-229 ec2-user]# 
```
Have few service containers running on the host.

From with in the host , reboot host using ```shutdown -r now``` .

When host reboots , host gets to "reconnecting" state and then back to "active" state as expected without any problems.

**Comment by sangeethah** [comment]
@panho66 , this fix will be available in the next  rancher-server release - v1.6.3

**Comment by panho66** [comment]
@sangeethah  Thanks :)

**Comment by Xaseron** [comment]
This problem still persists for me. Tested with rancher server 1.6.3 and 1.6.5 
On rhel 7.3 with native docker 1.12.6 

client docker info:
```
Containers: 9
 Running: 6
 Paused: 0
 Stopped: 3
Images: 10
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: vg01-thinpool
 Pool Blocksize: 524.3 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 1.011 GB
 Data Space Total: 24.15 GB
 Data Space Available: 23.14 GB
 Metadata Space Used: 520.2 kB
 Metadata Space Total: 532.7 MB
 Metadata Space Available: 532.2 MB
 Thin Pool Minimum Free Space: 2.415 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: bridge host null overlay
 Authorization: rhel-push-plugin
Swarm: inactive
Runtimes: runc docker-runc
Default Runtime: docker-runc
Security Options: seccomp
Kernel Version: 3.10.0-514.6.1.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 4
Total Memory: 7.64 GiB
Name: test
ID: S4NC:FORR:IPNT:YZMJ:VZ5L:UMIT:XE7R:2IWX:H2LX:7E3R:2URS:QSPT
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
Registries:  docker.io (secure)
```
